samples/Chaos/TodosClient.cs

```diff
@@ -0,0 +1,12 @@
+﻿namespace Chaos;
+
+public class TodosClient(HttpClient client)
+{
+    public async Task<IEnumerable<TodoModel>?> GetTodosAsync(CancellationToken cancellationToken)
+    {
+        using var request = new HttpRequestMessage(HttpMethod.Get, "/todos");
+        using var response = await client.SendAsync(request, cancellationToken);
```

Comment:
I wanted to include handling of invalid status codes which is not possible with that method.

Reply:
I thought that method implicitly called `EnsureStatusCode()` for you, so it's basically the same? I meant the one that's an overload of HttpClient.

Reply:
Based on my understanding it does not do status code checking. (it would prevent deserializing error bodies for example).

Reply:
Oh, if that's the case I've been using it wrong _a lot_ then. I assumed it checked the status code, as otherwise you'd get a weird result if you tried to deserialise the response as a model for an OK and instead the body was an ErrorDetails or an HTML error page or something.

Reply:
Also it would be weird to do a get knowing in advance you were just going to get an error details because it would fail. Not checking the status would make more sense if it was a discriminated union.

Reply:
You are right, it checks the status code as well. Simplified.